pkg/odo/cli/component/push.go

```diff
@@ -64,17 +65,58 @@ func NewPushOptions() *PushOptions {
 func (po *PushOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
 	po.DevfilePath = filepath.Join(po.componentContext, po.DevfilePath)
 
-	// if experimental mode is enabled and devfile is present
+	klog.V(4).Infof("Devfile path: %s", po.DevfilePath)
+
+	// If experimental mode is enabled and devfile is present
 	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(po.DevfilePath) {
-		envInfo, err := envinfo.NewEnvSpecificInfo(po.componentContext)
+		// We retrieve the configuration information. If this does not exist, then BLANK is returned (important!).
+		envFileInfo, err := envinfo.NewEnvSpecificInfo(po.componentContext)
 		if err != nil {
 			return errors.Wrap(err, "unable to retrieve configuration information")
 		}
-		po.EnvSpecificInfo = envInfo
+
+		// If the file does not exist, we should populate the environment file with the correct env.yaml information
+		// such as name and namespace.
+		// NOTE: Should we implement URL details later?
+		if !envFileInfo.EnvInfoFileExists() {
+			klog.V(4).Info("Environment file does not exist, creating the env.yaml file")
+
+			// Since the environment file does not exist, we will retrieve a correct namespace from
+			// either cmd commands or the current default kubernetes namespace
+			namespace, err := checkCmdNamespaceDefaultIfNotAvailable(cmd)
+			if err != nil {
+				return errors.Wrap(err, "unable to retrieve the default namespace for devfile")
+			}
+
+			// Retrieve a default name
+			// 1. Use args[1] if the user has supplied a name to be used
+			// 2. If the user did not provide a name, use gatherName to retrieve a name from the devfile.Metadata
+			// 3. Use the folder name that we are pushing from as a default name if none of the above exist
+			var name string
+			if len(args) == 2 {
+				name = args[1]
```

Comment:
It is not taking the args passed from cli, as maxargs here is one
https://github.com/openshift/odo/blob/77c876beef3f456067bff41b727914c6bf9a3725/pkg/odo/cli/component/push.go#L162

odo push my-component, creating component name from folder name, instead of passed name
```
[adisky@localhost nodejs-ex]$ odo push my-component

Validation
 ✓  Validating the devfile [42817ns]

Creating Kubernetes resources for component nodejs-ex
 ✓  Waiting for component to start [7s]

```